### PR TITLE
[mongo-c-driver/libbson] update to 1.25.0

### LINKED ISSUE #34855: mongo-c-driver update to 1.25.0
### Library name

mongo-c-driver

### New version number

1.25.0

### Other information that may be useful (release notes, etc...)

C driver 1.25.0 removes the source tarball from the release. Refer: [CDRIVER-4640](https://jira.mongodb.org/browse/CDRIVER-4640). If needed, the tarball produced by the GitHub release may be used as an alternative. Building from the tarball produced by GitHub requires specifying a `VERSION_CURRENT` file. The build requires a version. When git history is available, the version is computed. When building from the tarball the version needs to be specified.

C driver 1.25.0 removes the conditional dependency on libicu. Refer: [CDRIVER-4680](https://jira.mongodb.org/browse/CDRIVER-4680)

### PR DESCRIPTION
**Pull request by Adela0814**
Fixes https://github.com/microsoft/vcpkg/issues/34855
All feature tested pass on the following triplets:

- x64-windows
- x64-windows-static
- x86-windows
Usage tested pass on `x64-windows`.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/mongo-c-driver/portfile.cmake [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO mongodb/mongo-c-driver
     REF "${VERSION}"
-    SHA512 242416638bf6722ccda67ecc15cf78f8531f1a4ae0e89fd66cde74221d7b98b859442f5b70628802972961e69156fd7afcc916d7a83cc134d5723eb0b6e15cc9
+    SHA512 e35878eed6fe56a383eeb07795bcb6bfbaa29cd3d63c0efd67748e10982c9c1294c2de2127471b1f154ea658747c3a98e8d9f9a1cbcf1f5bf4ec6aa6e7b276a6
     HEAD_REF master
     PATCHES
-        disable-dynamic-when-static.patch
```
Comment (kevinAlbs) [inline ports/mongo-c-driver/portfile.cmake:12]: At present, there is no option to disable dynamic libmongoc (refer: [CDRIVER-4784](https://jira.mongodb.org/browse/CDRIVER-4784)). I suggest applying to add a patch to conditionally disable dynamic libmongoc: https://github.com/kevinAlbs/vcpkg/commit/b20bf0df13a70cf364c47c7e6fb7cf2e6f706c16

**Review by kevinAlbs — COMMENTED** [review]
Suggest applying commits from [this branch](https://github.com/kevinAlbs/vcpkg/commits/disabledynamic.Dev/Mengna/mongo-c-driver) to fix the `windows_static` tasks.

File: ports/libbson/portfile.cmake [hunk 2]
```diff
@@ -59,6 +54,10 @@ file(REMOVE_RECURSE
     "${CURRENT_PACKAGES_DIR}/share/mongo-c-driver"
 )
 
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
```
[the remark this answers is not in the document]
Comment (jimwang118) [reply ports/libbson/portfile.cmake:57]: Yes, now it is determined whether to compile a static or dynamic library through the external compilation options.

**Comment by jimwang118** [comment]
Duplicate of https://github.com/microsoft/vcpkg/pull/34994. This PR upgrades `libbson` to 1.25.1.

**Comment by dg0yt** [comment]
#34994 was merged.

**Pull request closed without merging**